libraries/opensk/src/api/persist.rs

```diff
@@ -232,27 +232,48 @@ pub trait Persist {
         self.insert(keys::MIN_PIN_LENGTH_RP_IDS, min_pin_length_rp_ids_bytes)
     }
 
-    // TODO rework LargeBlob
-    // Problem 1: Env should be allowed to choose whether to buffer in memory or persist
-    // Otherwise small RAM devices have limited large blog size.
-    // Problem 2: LargeBlob is a stateful command, but doesn't use the safeguard and infrastructure
-    // of Stateful command. It has to be migrated there.
-    // While doing that, check if PinUvAuthToken timers and StatefulCommand timeouts are working
-    // together correctly.
+    /// Prepares writing a new large blob.
+    ///
+    /// Returns a buffer that is returned to other API calls for potential usage.
+    fn init_large_blob(&mut self, expected_length: usize) -> CtapResult<Vec<u8>> {
+        Ok(Vec::with_capacity(expected_length))
+    }
+
+    /// Writes a large blob chunk to the buffer.
+    ///
+    /// This can be the passed in buffer, or a custom solution.
+    fn write_large_blob_chunk(
+        &mut self,
+        offset: usize,
+        chunk: &mut Vec<u8>,
```

Comment:
I don't understand why we pass this as `&mut Vec<u8>` instead of `&[u8]`. We don't reuse the allocation and `u8` is copy, so we don't need ownership. The `.append(` below would be `.extend_from_slice(`.

Reply:
The idea of `append` instead instead of `extend_from_slice` is to save memory. The former drains the `Vec`, the latter copies. Since this is a potentially memory sensitive operation, I think this optimization of restricting the type could be useful.

Reply:
That's what I meant by "we don't reuse the allocation", we don't save memory by using `append` because `u8` is copy and the initial vector won't be deallocated until everything is copied. If the input was `Vec<Box<Foo>>` then yes, `append` would reuse the `Foo` allocations (but still not reuse the `Vec` allocation).